tls/tls_srv.c

```diff
@@ -2160,6 +1993,23 @@ ttls_handshake_server_hello(TlsCtx *tls)
 		if ((r = ttls_write_server_hello(tls, &sgt, &p)))
 			T_FSM_EXIT();
 		CHECK_STATE(128);
+		if ((tls->state = TTLS_SERVER_CHANGE_CIPHER_SPEC)) {
```

Comment:
Do we really need an assignment here? That looks suspicious.
If for some reason this is what was intended, condition is always true, and need not to be checked.

Reply:
Yeah, that was a bug, there must be `if (tls->state == TTLS_SERVER_CHANGE_CIPHER_SPEC)`. Thank you!